.gitlab-ci.yml

```diff
@@ -91,6 +98,126 @@ xliff-lint:
     - xmllint --schema ./xliff-core-1.2-strict.xsd
       --noout $(find Resources -name '*.xlf')
 
+unit-tests-php7.2-t3-v10:
+  image: php:7.2
+  stage: test
+  needs:
+    - build-composer-dependencies
+  script:
+    - composer ci:tests:unit
+
+unit-tests-php7.3-t3-v10:
+  image: php:7.3
+  stage: test
+  needs:
+    - build-composer-dependencies
+  script:
+    - composer ci:tests:unit
+
+unit-tests-php7.4-t3-v10:
+  image: php:7.4
+  stage: test
+  needs:
+    - build-composer-dependencies
+  script:
+    - composer ci:tests:unit
+
+functional-tests-php7.2-t3-v10:
+  image: php:7.2
+  services:
+    - mariadb:10
+  stage: test
+  needs:
+    - build-composer-dependencies
+  script:
+    - composer ci:tests:functional
+
+functional-tests-php7.3-t3-v10:
+  image: php:7.3
+  services:
+    - mariadb:10
+  stage: test
+  needs:
+    - build-composer-dependencies
+  script:
+    - composer ci:tests:functional
+
+functional-tests-php7.4-t3-v10:
+  image: php:7.4
+  services:
+    - mariadb:10
+  stage: test
+  needs:
+    - build-composer-dependencies
+  script:
+    - composer ci:tests:functional
+
+unit-tests-php7.2-t3-v9:
+  image: php:7.2
+  stage: test
+  dependencies: []
```

Comment:
Is this required?

Reply:
it's required as dependencies are passed from stage to stage and if we have vendor directory, then downgrading to t3 v9 don't work correctly